xcmds/cpu/init.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This is init code for the case that cpu finds itself as pid 1.
+// This is duplicative of the real init, but we're implementing it
+// as a duplicate so we can get some idea of:
+// what an init package should have
+// what an init interface should have
+// So we take a bit of duplication now to better understand these
+// things. We also assume for now this is a busybox environment.
+// It is unusual (I guess?) for cpu to be an init in anything else.
+// So far, the case for an init pkg is not as strong as I thought
+// it might be.
+package main
+
+import (
+	"flag"
+	"log"
+	"os/exec"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/uroot/util"
+)
+
+var (
+	test     = flag.Bool("test", false, "Test mode: don't try to set control tty")
+	osInitGo = func() {}
+)
+
+func cpuSetup() error {
+	flag.Parse()
+	log.Printf("Welcome to Plan 9(tm)!")
+	util.Rootfs()
+	log.Printf("Done Rootfs")
+	osInitGo()
+	// TODO: this needs to be added as prt of the Rootfs() stuff
+	if o, err := exec.Command("ip", "link", "set", "dev", "lo", "up").CombinedOutput(); err != nil {
+		log.Fatalf("ip link set dev lo: %v (%v)", string(o), err)
+	}
+	return nil
+}
+
+func cpuDone() {
+	// We need to reap all children before exiting.
```

Comment:
The children are supposed to be reaped as they die, right? Not all at the end.

Reply:
yeah, I made a change, see what you think.